packages/dds/SchemaVersioning.md

```diff
@@ -0,0 +1,159 @@
+    This model risks the application authors not knowing about the pattern, so it requires clear communication of the approach.
+
+-   Removal of support for writing a format is a breaking change, and can only be done with a major version bump.
+
+    Since updating to a new format requires a full rollout (sometimes two if the format is not supported in the currently deployed version), and the library does not control this schedule,
+    dropping support for a format needs to be communicated to users of the library well in advance (more so than a simple API break that has no rollout requirements).
+    For Fluid Framework this means announcing the deprecation of the format as part of the major release before the one where it is removed.
+
+## Applications
+
+-   Applications should opt into writing new data formats only once support for those formats has been fully deployed. This includes enough testing to be confident they won't need to roll back to a version without the support.
+-   To best support Fluids maximum 30 day session length, applications may want to include an extra 30 days after the last rollout of support for the format before starting the rollout of enabling the new format for writing.
+-   When updating to a new major version of a library (such as Fluid), check the release notes for current or upcoming format changes, and schedule format updates accordingly.
+
+# Complications
+
+It’s worth discussing what happens if we do not follow these rules.
+A couple examples:
+
+## Deprecating DDS
+
+We decide to deprecate some DDS by eventually removing it from code base. Clients who rely on it will be stuck, and have one of these options:
+
+1. Stay forever on old versions of Fluid Framework.
+2. Copy DDS code into their repo and continue to support it the way it’s described at the beginning of the document.
+
+It’s a possibility, but it’s very inefficient if many customers depend on it, then I’d argue it’s better to push it to separate repo for partners to share cost of supporting it. The obvious next step – we continue to support it, which is not much different form having it in our repo under “legacy” bucket.
+
+Note that this workflow does not work with other changes, like changes in snapshot format.
+
+On-the fly converters
+
+In some distant future, we may offer converter workflow, where (for example) Directory DDS can be morphed transparently into SharedTree DDS on open / first edit (and API not exposing Directory DDS at all), including any trailing ops. Based on assumptions above, we will need to deploy such changes dark and eventually trigger conversion. Due to offline clients having local changes in old format, we will need to deal with merging ops in old and new format. Given that in general, better support for offline will require addition of 3-way merge, we are likely to rely on 3-way merge to deal with these changes (it simplifies certain aspects of conversion, but requires more expensive, but more generic merge tooling).
```

Comment:
```suggestion
In some distant future, we may offer a converter workflow, where (for example) a Directory DDS can be morphed transparently into a SharedTree DDS on open / first edit (and the API does not expose the Directory DDS at all), including any trailing ops.
Based on the assumptions above, we will need to deploy such changes dark and eventually trigger conversion.
Due to offline clients having local changes in the old format, we will need to deal with merging ops in the old and new formats.
Given that in general, better support for offline will require addition of 3-way merge, we are likely to rely on 3-way merge to deal with these changes (it simplifies certain aspects of conversion, but requires more expensive, but more generic merge tooling).
```

(or break more on commas if you want; hard to judge if the sentences are reasonable in this editor)